aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,785 @@
-.. include:: /../common/stub-topic.txt
+Razor Syntax Reference    
+===========================
+`Taylor Mullen <https://twitter.com/ntaylormullen>`__, and `Rick Anderson`_
 
-|stub-icon| Razor Syntax
-========================
+.. contents:: Sections
+  :local:
+  :depth: 1
+  
+What is Razor?
+--------------
+Razor is a markup syntax for embedding server based code into web pages. The Razor syntax consists of Razor markup, C# and HTML. Files containing Razor generally have a *.cshtml* file extension.
 
-.. include:: /../common/stub-notice.txt
+.. contents:: Sections:
+  :local:
+  :depth: 1
+  
+Rendering HTML
+--------------
 
-.. _issue: https://github.com/aspnet/Docs/issues/130
+The default Razor language is HTML. Rendering HTML from Razor is no different than in an HTML file. A Razor file with the following markup:
+
+.. code-block:: none
+
+  <p>Hello World</p> 
+
+Is rendered unchanged as ``<p>Hello World</p>`` by the server.
+
+Rendering server code
+----------------------
+
+Razor supports C# and uses the ``@`` symbol to transition from HTML to C#. 
+Razor can transition from HTML into C# or into Razor specific markup. When an ``@`` symbol is followed by a :ref:`Razor reserved keyword <Razor-reserved-keywords-label>` it transitions into Razor specific markup, otherwise it transitions into plain C# . 
+
+Razor expressions
+---------------------
+
+Implicit Razor expressions start with ``@`` followed by C# code. Explicit Razor expressions start with ``@`` and are in a block enclosed by ``()`` or ``{}``. For example:
```

Comment:
`@{}` is not an expression. It's a code block. The below inclusions should not have `@{}`.